src/pages/iv-items/forms/SaleList.js

```diff
@@ -0,0 +1,289 @@
+import * as yup from 'yup'
+
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import { useForm } from 'src/hooks/form.js'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { Grid } from '@mui/material'
+import SalesForm from './SalesForm'
+import FormShell from 'src/components/Shared/FormShell'
+
+const SalesList = ({ store, labels, maxAccess, formikInitial }) => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { recordId } = store
+  const { _msId } = store
+  const { measurementId } = store
+  const { priceGroupId } = store
+  const { returnPolicy } = store
+
+  const { platformLabels } = useContext(ControlContext)
+
+  const { stack } = useWindow()
+
+  const [check, setCheck] = useState(false)
+  const [firstCurr, setFirstCurr] = useState(false)
+  useEffect(() => {
+    if (recordId) {
+      const fetchCurrency = async () => {
+        try {
+          const response = await getRequest({
+            extension: InventoryRepository.Currency.qry,
+            parameters: `&_itemId=${recordId}`
+          })
+          if (response.list && response.list.length > 0) {
+            setFirstCurr(response.list[0].currencyId)
+          }
+        } catch (error) {}
+      }
+
+      fetchCurrency()
+    }
+  }, [])
+
+  const { formik } = useForm({
+    initialValues: {
+      currencyId: firstCurr || '',
+      defSaleMUId: measurementId || '',
+      pgId: priceGroupId || '',
+      returnPolicyId: returnPolicy || ''
+    },
+
+    enableReinitialize: true,
+    validateOnChange: true,
+    onSubmit: async obj => {
+      const submissionData = {
+        ...formikInitial,
+        defSaleMUId: formik.values.defSaleMUId,
+        pgId: formik.values.pgId,
+        returnPolicyId: formik.values.returnPolicyId
+      }
+
+      const response = await postRequest({
+        extension: InventoryRepository.Items.set,
+        record: JSON.stringify(submissionData)
+      })
+    }
+  })
+
+  async function fetchGridData() {
+    if (formik.values.currencyId) {
+      const response = await getRequest({
+        extension: SaleRepository.Sales.qry,
+        parameters: `&_itemId=${recordId}&_currencyId=${formik.values.currencyId}`
+      })
+
+      return response
+    }
+  }
+
+  const {
+    query: { data },
+    labels: _labels,
+    invalidate,
+
+    refetch
+  } = useResourceQuery({
+    enabled: !!recordId,
+    datasetId: ResourceIds.SalesList,
+    queryFn: fetchGridData,
+    endpointId: SaleRepository.Sales.qry
+  })
+
+  useEffect(() => {
+    if (formik.values.currencyId) {
```

Comment:
You are checking formik.values.currencyId twice , first inside useEffect and the second inside fetchGridData. Remove one of them.